react/features/transcription/components/TranscriptionLabel.web.js

```diff
@@ -0,0 +1,145 @@
+ * the transcription current state.
+ */
+export class TranscriptionLabel extends Component {
+    /**
+     * {@code VideoQualityLabel}'s property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * Whether or not the filmstrip is currently set to be displayed.
+         */
+        _filmstripVisible: React.PropTypes.bool,
+
+        /**
+         * Whether or not remote videos within the filmstrip are currently
+         * visible. Depending on the visibility state, coupled with filmstrip
+         * visibility, CSS classes will be set to allow for adjusting of
+         * {@code TranscriptionLabel} positioning.
+         */
+        _remoteVideosVisible: React.PropTypes.bool,
+
+        /**
+         * The current state of the transcription.
+         */
+        _transcriptionState: React.PropTypes.string
+    };
+
+    /**
+     * Initializes a new {@code TranscriptionLabel} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            /**
+             * Whether or not the filmstrip was not visible but has transitioned
+             * in the latest component update to visible. This boolean is used
+             * to set a class for position animations.
+             *
+             * @type {boolean}
+             */
+            filmstripBecomingVisible: false
+        };
+    }
+
+    /**
+     * Updates the state for whether or not the filmstrip is being toggled to
+     * display after having being hidden.
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The read-only props which this Component will
+     * receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        this.setState({
+            filmstripBecomingVisible: nextProps._filmstripVisible
+            && !this.props._filmstripVisible
+        });
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const isVisible = this.props._transcriptionState
+            === JitsiMeetJS.constants.transcriptionStatus.ON;
+
+        const rootClassName = [
+            'video-state-indicator moveToCorner',
+            isVisible ? 'show-inline' : '',
+            this.state.filmstripBecomingVisible ? 'opening' : '',
+            this.props._filmstripVisible
+                ? 'with-filmstrip' : 'without-filmstrip',
+            this.props._remoteVideosVisible
+                ? 'with-remote-videos' : 'without-remote-videos'
+        ].join(' ');
+
+        return (
+            isVisible
+                ? <span
+                    className = { rootClassName }
+                    id = 'transcriptionLabel'>
+                    <span className = 'icon-edit' />
```

Comment:
AtlasKit has a similar icon. Maybe that could be used instead or is it improper usage?

Reply:
If we use the AtlasKit one, the proper way is to use the react component for it and we currently still use the font icons, because they work for mobile as well.